core/src/main/java/com/graphhopper/routing/HeadingResolver.java

```diff
@@ -59,16 +60,46 @@ public IntArrayList getEdgesWithDifferentHeading(int baseNode, double heading) {
         return edges;
     }
 
+    /**
+     * This method returns true if the specified heading is parallel to the specified edgeState (antiparallel isn't tested).
+     * Note that only the road segments near the specified pointNearHeading are checked for parallelism (<20m) and that
+     * a angle difference of 30° is accepted.
+     */
+    public static boolean isHeadingNearlyParallel(EdgeIteratorState edgeState, double heading, GHPoint pointNearHeading) {
+        DistanceCalc calcDist = DistanceCalcEarth.DIST_EARTH;
+        double xAxisAngle = AngleCalc.ANGLE_CALC.convertAzimuth2xaxisAngle(heading);
+        PointList points = edgeState.fetchWayGeometry(FetchMode.ALL);
+        int closestPoint = -1;
+        // TODO should we use the same default like for gpx_accuracy which is 40m?
```

Comment:
I'd say yes because the heading is most likely reported from a device using GPS data.

Reply:
Yes, the data also comes from GPS. Still the use case is different IMO (not 100%). For map matching the number of candidates increases with a bigger gpx_accuracy which could lead to a more precise overall match (let's neglect performance for this). For heading the quality can decrease arbitrarily when we increase this too high (as we reject closer roads with a different heading). Let's assume the heading is also picked from the device than it might be bad for slow speed (ok, we could argue that the application could determine these bad situations and remove the heading parameter)

I propose to use a different value as a start (and increase this to e.g. 30m and later 40m if not sufficient in practice).

Reply:
> For heading the quality can decrease arbitrarily when we increase this too high (as we reject closer roads with a different heading).

Isn't that the point of the heading parameter?

> Let's assume the heading is also picked from the device than it might be bad for slow speed (ok, we could argue that the client could determine these bad situations and remove the heading parameter)

Bad GPS precision is far more likely e.g if the signal is reflected/blocked

It would be nice if the heading parameter would be used to prioritize fitting edges and not enforced as a hard filter.